source/fab/builder.py

```diff
@@ -119,43 +109,67 @@ def __init__(self,
             workspace.mkdir(parents=True)
 
         self._state = SqliteStateDatabase(workspace)
-        self._target = target
-        self._exec_name = exec_name
-
-        self._command_flags_map: Dict[Type[Command], List[str]] = {}
-        if fpp_flags != '':
-            self._command_flags_map[FortranPreProcessor] = (
-                fpp_flags.split()
-            )
-        if fc_flags != '':
-            self._command_flags_map[FortranCompiler] = (
-                fc_flags.split()
-            )
-        if ld_flags != '':
-            self._command_flags_map[FortranLinker] = (
-                ld_flags.split()
-            )
-        self._queue = QueueManager(n_procs - 1)
 
-    def _extend_task_queue(self, task: Task) -> None:
-        self._queue.add_to_queue(task)
-        for prereq in task.prerequisites:
-            self._queue.add_to_queue(HashCalculator(FileTextReader(prereq),
-                                                    self._state))
+        # If not given, use the name of the target as the exec name
+        # TODO: Should this be in the CLI part?
```

Comment:
I think it probably should. The actual implementation of the business logic can expect to be given exact details of what to do. It is the UIs job to deal with default behavior and stuff like that. Please create a ticket for moving this or do it now if the opportunity presents itself.